crates/bevy_pbr/src/render/mesh_view_types.wgsl

```diff
@@ -61,6 +61,31 @@ struct Lights {
     spot_light_shadowmap_offset: i32,
 };
 
+struct Fog {
+    mode: u32,
+    base_color: vec4<f32>,
+    directional_light_color: vec4<f32>,
+    directional_light_exponent: f32,
+    // `be` and `bi` are allocated differently depending on the fog mode
+    //
+    // For Linear Fog:
+    //     be.x = start, be.y = end
+    // For Exponential and ExponentialSquared Fog:
+    //     be.x = density
+    // For Atmospheric Fog:
+    //     be = per-channel extinction density
+    //     bi = per-channel inscattering density
+    be: vec3<f32>,
+    bi: vec3<f32>,
+}
+
+// Important: These must be kept in sync with `fog.rs`
+let FOG_MODE_OFF: u32                   = 0u;
+let FOG_MODE_LINEAR: u32                = 1u;
+let FOG_MODE_EXPONENTIAL: u32           = 2u;
+let FOG_MODE_EXPONENTIAL_SQUARED: u32   = 3u;
+let FOG_MODE_ATMOSPHERIC: u32           = 4u;
```

Comment:
If I read #7356 correctly, these let must be changed to const once it i merged.

Reply:
Thanks for the heads up, will keep an eye out.